arm/cores/wiring_analog.c

```diff
@@ -186,66 +241,87 @@ return -1;
 }
 
 
-void analogWrite( uint8_t pin, uint16_t value )
```

Comment:
Arduino Reference Language compatibility needs to be ensured. This change breaks its. All API prototypes must be preserved.

Reply:
The Arduino Refernece is a moving target and this does **NOT** break anything as  the ONLY situations this could break anything are NOT supported by Arduino IDE
- Modules compiled in different language (e.g. Forth, Ruby, Java)) and the stubs/wrappers referenced a different version.
- You were able to link in a precompiled binary blob, as the licensing and way of building is everything this method cannot be used

Please remember that 99.999% of users do not realise that member functions like write and Print return a value and ignore it. Alo 99.9% do not know how to use return error codes, this gives even for support the ability to get

Many more users like to have error returns to work out why a call did not work.

If the users does not use the returned value it breaks nothing.

There are many things in the Arduino Reference that are reminiscent of a 90's thesis student project that assumes everything alweys works. Error returns HELP users

You are making lots of assumptions based on ONE board like I have seen before see #95 about assumptions all boards have analog pins in ascending order for port pins

Reply:
Understood and agree on the advantages of return codes. Nevertheless, I would not open the door to changes in the Arduino Reference Language core functions, and ensure that sketches are building and running for any compatible platforms. Sketches using the return value of such will not compile for other board families.

Reply:
Whilst I appreciate yor concern many things do not compile acroos Arduino's boards as many are incompatible. Very few boards from Arduino or other makers are compatible .

Until about 18 months ago there was many debates even on Arduino developers m,ailing list about the Arduino Reference stating that **_char_** was unsigned and many examples assumed it was AVR standard of **unsigned** and many SAM/ARM etc boards where **_char_** is **signed**. More recently they added the **_byte_**  datatype and updated the reference

The ways of defining an INPUT_PULLUP varies across Arduino's own boards on AVR you do digitalWrite before PinMode to achieve this all others support INPUT_PULLUP. Other input types vary across boards even in same family.

Let alone the variations on attachinterrupt as the hardware is not compatible or easily able to be one or some cases multiple combinations of types like

- Interrupt
- Change
- High
- Low
- Rising
- Falling

There are other inconsistencies around software PWM, interrupt pins and other features like implementation of SPI.

Whilst your goal is noble I think it is restrictive unnecessarily on a moving target

Reply:
Further proof that others use return codes that are errors see
https://www.espruino.com/EspruinoESP8266#analogread-implementation

Espruino ESP8266 popular wifi unit Arduino compatability
~~~
The esp8266 ADC function is available on pin A0 or without a pin. Using a digital pin will return NaN

Possible ways to call ananlogRead() and the returns:

analogRead(A0);
=0.0029296875

analogRead();
=0.0029296875

analogRead(D0);
=NaN
~~~

Reply:
Good and valid point :) In your last example (don´t know the rest) there is a fundamental difference with our xmc-for-arduino: they are pointing out the differences and have documented their implementation details. 

I am an idealist xD and the goal should be to provide a certain level of interoperability, ease-of-use, and the standarization of a common ecosystem. Does it means that the reference language is shared among all the cores and their function prototypes preserved?  Does this just mean that the Arduino cores are just enabled to be used with the Arduino IDE? Or even just a hardware connectivity compatibility? 
These dicussions of 1-to-1 compatibility of built-in libraries and reference language can be endless 👯‍♂ Let´s not make a big deal out of this minor changes (at least for now).

The main issue is to have a board agnostic implementation, try to keep the project maintanable and ensure the features keep working properly with existing and new boards.

Reply:
In an ideal world HardwareSerial would not exist as a core module (and actually could have been achived). Many things are undocumented in the Arduino core like on Serial receive Framing and Overrun errors put grabge data into buffers with no error flags or counts, whilst Pairty Error discards data SILENTLY. Some variations and suggestions have added 8 bit counters to at least count errors.

The XMC-for-Arduino differences to what people **think of as the core reference** are documented not necessarily on this repository or its wiki. I do not remember if I have edit/write permissions to wiki. However for XMC1300 Boot kit on this Wiki points to my Wiki.

For existing differences between _Reference_ and XMC-for-Arduino please see the following pages on my repository Wiki and note some are not changeable to match without MAJOR work

[Functions GPIO](https://github.com/techpaul/XMC-for-Arduino/wiki/Functions-and-Options-Differences)
[Analog functions and improvements](https://github.com/techpaul/XMC-for-Arduino/wiki/Analog-Functions-and-Additions) 

Note some of those changes enable the IFX92101 support to work correctly